packages/js/components/src/experimental-select-tree-control/select-tree.tsx

```diff
@@ -0,0 +1,221 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/**
+ * External dependencies
+ */
+import { createElement, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { search } from '@wordpress/icons';
+import { Dropdown, Spinner } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { useLinkedTree } from '../experimental-tree-control/hooks/use-linked-tree';
+import { Tree } from '../experimental-tree-control/tree';
+import {
+	Item,
+	LinkedTree,
+	TreeControlProps,
+} from '../experimental-tree-control/types';
+import { SelectedItems } from '../experimental-select-control/selected-items';
+import {
+	getItemLabelType,
+	getItemValueType,
+} from '../experimental-select-control/types';
+import {
+	defaultGetItemLabel,
+	defaultGetItemValue,
+} from '../experimental-select-control/utils';
+import { ComboBox } from '../experimental-select-control/combo-box';
+import { SuffixIcon } from '../experimental-select-control/suffix-icon';
+
+interface SelectTreeProps extends Omit< TreeControlProps, 'getItemLabel' > {
+	selected?: Item[];
+	getItemLabel?: getItemLabelType< Item >;
+	getItemValue?: getItemValueType< Item >;
+	getSelectedItemProps?: any;
+	getFilteredItems: (
+		allItems: Item[],
+		inputValue: string,
+		selectedItems: Item[],
+		getItemLabel: getItemLabelType< Item >
+	) => any[];
+	treeRef?: React.ForwardedRef< HTMLOListElement >;
+	suffix?: JSX.Element | null;
+	isLoading?: boolean;
+	label: string | JSX.Element;
+	onInputChange?: ( value: string | undefined ) => void;
+}
+
+export const SelectTree = function SelectTree( {
+	items,
+	getItemLabel = ( item ) => item?.name || '',
+	getItemValue = ( item ) => item?.id || '',
+	getSelectedItemProps,
+	getFilteredItems,
```

Comment:
This function introduces another layer of complexity inside the component at render time. What `items` prop is for then? Filtering items should be a responsibility managed outside of the component like in the server.

Reply:
I changed the implementation. One upside of always having all the items inside SelectTree and Tree controls would be to implement reusable logic, but I think it's fine to switch it for now.